src/utilities/utilities.js

```diff
@@ -47,6 +63,90 @@ export function check_standard_compliance(standard, kwargs) {
   }
 }
 
+/**
+ * @typedef {Object.<string, number[]>} CheckStandardComplianceResult
+ * @property {number[]} tdb
+ * @property {number[]} tr
+ * @property {number[]} v
+ * @property {number[]} [met]
+ * @property {number[]} [clo]
+ * @property {number[]} [rh]
+ */
+
+/**
+ * Check that the values as an array comply with the standard provided and returns arrays where
+ * the values that do not comply are NaN
+ * @see {@link check_standard_compliance} for scalar variant that returns warnings
+ *
+ * @param {Standard | "fan_heatwaves"} standard - standard to check compliance with
+ * @param {ComplianceKwargsArray & {airspeed_control?: boolean}} kwargs - values to check compliance against
+ *
+ * @returns {CheckStandardComplianceResult} filtered arrays based on compliance limits
+ */
+export function check_standard_compliance_array(standard, kwargs) {
```

Comment:
This looks _really_ long, and I know I wouldn't be confident changing it unless I had some tests to assure me that I hadn't broken anything along the way.

Reply:
yep, will add them